packages/backend-core/src/sql/sql.ts

```diff
@@ -371,10 +373,11 @@ class InternalBuilder {
             ),
             castedTypeValue.values
           )
-        } else if (!opts?.relationship && !isRelationshipField) {
+        } else if (!isRelationshipField) {
           const alias = getTableAlias(tableName)
           fn(alias ? `${alias}.${updatedKey}` : updatedKey, value)
-        } else if (opts?.relationship && isRelationshipField) {
+        }
+        if (opts?.relationship && isRelationshipField) {
```

Comment:
Why removing this else?

Reply:
This is somewhat of a hack for now - basically if there are `$or` relationship filters we need to have both filters below here - it depends on the logic, but previously we only had the filters for relationships here, then we did the filters for the main table in the sub-query.

This is a short term fix as the changes in https://github.com/Budibase/budibase/pull/14410 completely remove the need for this - technically it could be removed since once the new pagination mechanism is merged is completely fixes this problem.